### PR TITLE
[PM-3543] [PM-3607] Fix password re-prompt when editing and on autofill.

### PR DESCRIPTION
**Pull request by andrebispo5**
## Type of change
- [X] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

Fix 2 bugs on password re-prompt. On autofill and editing a cipher, the app would crash.

## Code changes

Update code to use userkey.

## Before you submit
- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/App/Services/MobilePasswordRepromptService.cs [hunk 1]
```diff
@@ -42,7 +47,37 @@ private async Task<bool> ValidatePasswordAsync(string password)
                 return false;
             };
 
-            return await _cryptoService.CompareAndUpdateKeyHashAsync(password, null);
+            var email = await _stateService.GetEmailAsync();
+            var kdfConfig = await _stateService.GetActiveUserCustomDataAsync(a => new KdfConfig(a?.Profile));
+            var masterKey = await _cryptoService.MakeMasterKeyAsync(password, email, kdfConfig);
+
+            var storedPasswordHash = await _cryptoService.GetMasterKeyHashAsync();
+            if (storedPasswordHash == null)
```
Comment (jlf0dev) [inline src/App/Services/MobilePasswordRepromptService.cs:55]: We built a method in `clients` on the crypto service to `getOrDeriveMasterKey`. Can you copy [that](https://github.com/bitwarden/clients/blob/5665576147a4fbf30024f56358d424ff26a1a131/libs/common/src/platform/services/crypto.service.ts#L161-L169) logic here?

File: src/App/Services/MobilePasswordRepromptService.cs [hunk 2]
```diff
@@ -42,7 +47,37 @@ private async Task<bool> ValidatePasswordAsync(string password)
...
+            {
+                var oldKey = await _secureStorageService.GetAsync<string>("oldKey");
+                if (masterKey.KeyB64 == oldKey)
+                {
+                    var localPasswordHash = await _cryptoService.HashMasterKeyAsync(password, masterKey, HashPurpose.LocalAuthorization);
+                    await _secureStorageService.RemoveAsync("oldKey");
+                    await _cryptoService.SetMasterKeyHashAsync(localPasswordHash);
```
Comment (jlf0dev) [inline src/App/Services/MobilePasswordRepromptService.cs:62]: @mpbw2 Do you remember why we have this `oldKey`? It seems like we could probably remove support for it at this point.

**Review by jlf0dev — CHANGES_REQUESTED** [review]

File: src/App/Services/MobilePasswordRepromptService.cs (hunk 2 above)
Comment (mpbw2) [reply src/App/Services/MobilePasswordRepromptService.cs:62]: @jlf0dev It was from a migration that took place before I joined, so I agree it's probably safe to let it go.

Comment (andrebispo5) [reply src/App/Services/MobilePasswordRepromptService.cs:62]: Awesome 👍

**Review by jlf0dev — APPROVED** [review]

**Comment by equetzal** [comment]
Hey any new on when are we expecting to receive the fix update? 
Idk about other users but lot of us are relying and paying Bitwarden and this feature it's really needed to be deployed ad soon as possible (as a hot fix not in the regular deploy schedule).

**Pull request merged**